EZYVideoPlayer/Classes/EZYVideoPlayerModel.swift

```diff
@@ -0,0 +1,156 @@
+//
+//  EZYVideoPlayerModel.swift
+//  EZYVideoPlayer
+//
+//  Created by Shashank Pali on 09/01/23.
+//
+
+import Foundation
+import AVFoundation
+
+internal final class EZYVideoPlayerModel: NSObject, EZYVideoPlayerModelProtocol {
+    
+    var shouldPlay = true //default player will start playing when buffered
+    var timeObserver: Any?
+    weak var player: AVPlayer?
+    weak var delegate: EZYVideoPlayerDelegate? {
+        didSet {
+            delegate?.didChangedPlayer(status: .buffering)
+        }
+    }
+    
+    required init(urlString: String) {
+        super.init()
+        
+        guard let url = URL(string: urlString) else {return}
+        let asset = AVAsset(url: url)
+        let requiredAssetKeys = ["playable", "hasProtectedContent"]
+        let playerItem = AVPlayerItem(asset: asset, automaticallyLoadedAssetKeys: requiredAssetKeys)
+        
+        let p = AVPlayer(playerItem: playerItem)
+        self.player = p
+        observePlayer()
```

Comment:
Can we apply the change `self.player = AVPlayer(playerItem: playerItem)`?

Reply:
https://github.com/shashankpali/EZYVideoPlayer/pull/1#discussion_r1091756193